llvm/lib/IR/Verifier.cpp

```diff
@@ -399,6 +400,7 @@ class Verifier : public InstVisitor<Verifier>, VerifierSupport {
   bool hasBrokenDebugInfo() const { return BrokenDebugInfo; }
 
   bool verify(const Function &F) {
+    llvm::TimeTraceScope timeScope("Verifier");
```

Comment:
Maybe this is enough; do you really need the "Dominator Tree Builder" and "Verifier visit' timers as well?